packages/compass-e2e-tests/helpers/commands/select-connection-menu-item.ts

```diff
@@ -7,17 +8,35 @@ export async function selectConnectionMenuItem(
   itemSelector: string
 ) {
   const selector = Selectors.sidebarFavorite(favoriteName);
-  // It takes some time for the favourites to load
-  await browser.$(selector).waitForDisplayed();
 
-  // workaround for weirdness in the ItemActionControls menu opener icon
-  await browser.clickVisible(Selectors.ConnectionsTitle);
+  await browser.waitUntil(async () => {
+    console.log('.');
+    if (
+      await browser
+        .$(Selectors.sidebarFavoriteMenuButton(favoriteName))
+        .isDisplayed()
+    ) {
+      return true;
+    }
 
-  // Hover over an arbitrary other element to ensure that the second hover will
-  // actually be a fresh one. This otherwise breaks if this function is called
-  // twice in a row.
-  await browser.hover(`*:not(${selector}, ${selector} *)`);
-  await browser.hover(selector);
+    // It takes some time for the favourites to load
+    await browser.$(selector).waitForDisplayed();
+
+    // workaround for weirdness in the ItemActionControls menu opener icon
+    await browser.clickVisible(
+      TEST_MULTIPLE_CONNECTIONS
+        ? Selectors.SidebarHeader
+        : Selectors.ConnectionsTitle
+    );
+
+    // Hover over an arbitrary other element to ensure that the second hover will
+    // actually be a fresh one. This otherwise breaks if this function is called
+    // twice in a row.
+    await browser.hover(`*:not(${selector}, ${selector} *)`);
+
+    await browser.hover(selector);
+    return false;
+  });
 
   await browser.clickVisible(Selectors.sidebarFavoriteMenuButton(favoriteName));
   await browser.$(Selectors.ConnectionMenu).waitForDisplayed();
```

Comment:
since we still have the split connection lists, doesn't it matter if connection menu item is active or not? (actually even after uniting the lists, we'll likely care about this status as they don't have the same actions)

Reply:
ah, found the comment about this working with saved connections only, that answers my questions!